applications/StructuralMechanicsApplication/tests/test_custom_scipy_base_solver.py

```diff
@@ -0,0 +1,149 @@
+from __future__ import print_function, absolute_import, division
+import KratosMultiphysics
+
+import KratosMultiphysics.StructuralMechanicsApplication as StructuralMechanicsApplication
+from KratosMultiphysics.StructuralMechanicsApplication.structural_mechanics_analysis import StructuralMechanicsAnalysis
+import KratosMultiphysics.KratosUnittest as KratosUnittest
+from KratosMultiphysics import kratos_utilities as kratos_utils
+eigensolvers_application_available = kratos_utils.CheckIfApplicationsAvailable("EigenSolversApplication")
+
+'''
+Test description:
+This test does an eigenvalue analysis on a simple cantilever beam
+It compares the results between the scipy solver and the standard eigen solver.
+'''
+
+@KratosUnittest.skipUnless(eigensolvers_application_available,"Missing required application: EigenSolversApplication")
+class TestCustomScipyBaseSolver(KratosUnittest.TestCase):
+    # muting the output
+    KratosMultiphysics.Logger.GetDefaultOutput().SetSeverity(KratosMultiphysics.Logger.Severity.WARNING)
+
+    def test_eigen_with_constraints(self):
+        analysis_parameters_scipy = KratosMultiphysics.Parameters("""{
+            "problem_data"    : {
+                "parallel_type" : "OpenMP",
+                "echo_level"    : 1,
+                "start_time"    : 0.0,
+                "end_time"      : 1.0
+            },
+            "solver_settings" : {
+                "solver_type"              : "structural_mechanics_custom_scipy_base_solver",
```

Comment:
Please note: Test will fail. As I just provide the name of the solver to suit #6250.